### PR TITLE
Vercel migration

### PR DESCRIPTION
**Pull request by tony-sull**
Migrating the site from Netlify to Vercel

TODO:
- disable `main` build and deploy in Netlify before merging
- update Vercel project to deploy from `main`
- final test on Vercel temporary URL
- update DNS to point the production site to the new builds

### CODE REVIEW

File: netlify/functions/nightly-build.ts
Comment (tony-sull) [inline netlify/functions/nightly-build.ts:1]: not a great way to replicate this in Vercel unfortunately, I'll setup a [cron-job.org](https://cron-job.org) account to handle triggering a build webhook

File: src/utils/github.ts [hunk 1]
```diff
@@ -18,15 +18,18 @@ export type Example = {
 
 export function githubRequest(url: string) {
 	const headers: Headers = new Headers({
-		Accept: 'application/vnd.github.v3+json',
-	});
-	if (typeof import.meta.env.PUBLIC_VITE_GITHUB_TOKEN !== 'string') {
+		Accept: "application/vnd.github.v3+json",
+	})
+	if (typeof import.meta.env.GITHUB_TOKEN !== "string") {
```
Comment (tony-sull) [inline src/utils/github.ts:23]: we had a mix of a couple variables here, simplified to a single server-only variable

File: package.json [hunk 2]
```diff
@@ -25,22 +25,22 @@
   },
   "prettier": "@astrojs/site-kit/prettier",
   "dependencies": {
-    "@astrojs/netlify": "^2.2.2",
-    "@fontsource/inter": "^4.5.15",
+    "@astrojs/vercel": "^5.0.1",
+    "@fontsource-variable/inter": "^5.0.8",
+    "@fontsource/inter": "^5.0.8",
```
Comment (itsMapleLeaf) [inline package.json:30]: do we need both?

File: package.json [hunk 3]
```diff
@@ -25,22 +25,22 @@
...
     "title": "^3.5.3"
   },
   "devDependencies": {
-    "@astrojs/site-kit": "github:withastro/site-kit",
-    "@astrojs/tailwind": "^3.1.1",
-    "@netlify/functions": "^1.4.0",
+    "@astrojs/site-kit": "github:withastro/site-kit#v5.0.0",
```
Comment (itsMapleLeaf) [inline package.json:34]: This actually shouldn't be version locked 😅 the 5.0.0 tag is old. The lockfile ensures this doesn't get randomly upgraded unless we explicitly want that

File: package.json (hunk 2 above)
Comment (tony-sull) [reply package.json]: good catch, old one removed

File: package.json (hunk 3 above)
Comment (tony-sull) [reply package.json]: ah ha, ok that also explains all the random headaches I was having here with linting 😅 A day is never complete until I make at least one commit complaining about linters/formatters

**Review by itsMapleLeaf — APPROVED** [review]

**Pull request merged**